tests/components/trace/test_websocket_api.py

```diff
@@ -552,14 +552,24 @@ def next_id():
     await hass.async_block_till_done()
 
     # List traces
-    await client.send_json({"id": next_id(), "type": "trace/list"})
+    await client.send_json({"id": next_id(), "type": "trace/list", "domain": "script"})
     response = await client.receive_json()
     assert response["success"]
-    assert len(response["result"]) == 2
-    assert len(_find_traces(response["result"], domain, "sun")) == 1
+    if domain == "automation":
+        assert len(response["result"]) == 1
+    else:
+        assert len(response["result"]) == 2
     assert len(_find_traces(response["result"], "script", "moon")) == 1
-    sun_run_id = _find_run_id(response["result"], domain, "sun")
     moon_run_id = _find_run_id(response["result"], "script", "moon")
+    if domain == "automation":
```

Comment:
Using if/else logic in the tests together with parameters is not the best way to do test parametrization. Logic in tests is bad for both readability and robustness.

If it's not possible to parametrize and avoid the logic it's probably better to have separate tests for the different cases.

Reply:
Good point, I'll refactor to pass as parameters instead in a follow-up.